### ISSUE TITLE
System name label showing when the system isn't/shouldn't be

### ISSUE THREAD

**Issue by HammerGS** [MegaMek/mekhq#81]
This is an immersion issue more than an a major bug.  In the screen shot which is take during the Reunification war numerous system names are showing when the graphic for the system isn't.

Would be nice to have those names blanked out or not showing.  Also clutters the map less in some eras.

![image](https://cloud.githubusercontent.com/assets/13752452/15457144/47bf6e78-2041-11e6-899b-186431f260f4.png)

**Comment by Akjosch** [comment]
For that, I'd like to first have a proper split between

* systems which are *properly* undiscovered (your random M-class dwarf with a "forgotten" colony)

 -> not appearing on the map, not used for jump paths
* discovered (via telescopes or even because they were previously settled) and named but without proper navigational data set (at this point in time)

 -> appearing on the map, but not used for jump paths
* discovered and with proper navigational data set

 -> appearing on the map and used for jump paths
* unimportant for the map and unclaimed (the generic "NPwhatever" systems, for example)

 -> not appearing on the map, but used for jump paths

... and have it be independent from faction. Otherwise, we're just adding metadata to the faction identifier which has no business being there (the "undiscovered", "abandoned" and "independent" semi-factions already present have really no business being there and are just making campaign code bloated and hard to follow).

**Comment by HammerGS** [comment]
So really this can't happen till the planets.xml is sliced up into those proposed components.

We can't just hide the names so we don't look like we have names hanging in space?

**Comment by Akjosch** [comment]
The biggest issue is as usual how to style these Java Swing elements nicely.

![ismap1](https://cloud.githubusercontent.com/assets/1029805/15728965/58c8e7d6-2860-11e6-9ab8-e1b680914aa4.png)

Other then that, I should have something ready to go by this week-end. Likely with place for more display options if we feel like adding them.

**Comment by pheonixstorm** [comment]
I like having abandoned/undiscovered systems to jump through.Gives me crazy ownership ideas, plus a lot of those that are on the map but undiscovered are known but never colonized for whatever reason. And forgotten colonies would still have a star to path through but not a planet worth mentioning.
Just my dozen or so pennies.

**Comment by Akjosch** [comment]
A simple option to show/hide the names of "abandoned", "undiscovered" and "unexplored" systems added in https://github.com/MegaMek/mekhq/commit/0f8dc1cc66bb0698995bc433be56dc12d4604e74

They are still used for pathing/jumping through and are still selectable (and still technically rendered as black dots).

**Comment by GustavKuriga** [comment]
Kind of related, would it be possible to change the faction ownership of a
system in-program? This would save a lot of fuss for GMs running campaigns
where the players can effect such things.

**Comment by pheonixstorm** [comment]
I think that is in the current release

**Comment by Akjosch** [comment]
It is indeed. Turn on GM Mode, mark the planet on the map, right-click and pick "Edit planetary events for (planet)" from the GM Tools.

**Comment by Akjosch** [comment]
So, any feedback on the feature as currently implemented?

**Comment by HammerGS** [comment]
So far so good.  The hiding of the names works great for my purposes and haven't noticed anything odd.

**Comment by pheonixstorm** [comment]
@HammerGS @Akjosch do both of you feel this is complete and ready to close?

**Comment by Akjosch** [comment]
Yeah, looks like it.
